### PR TITLE
Add Psychic document loader

### PR DESCRIPTION
**Pull request by Ayan-Bandyopadhyay**
# Psychic document loader

Psychic.dev is a universal API to connect to your users' SaaS apps such as Notion, Google Drive, and Zendesk. Unlike other document loaders, Psychic handles the entire frontend OAuth flow, and provides a universal API to query data from all SaaS apps. This PR adds the Psychic document loader to Langchain so that developers can use the data locked away in SaaS tools for Document Retrieval QA. 

This PR adds the psychic document loader along with documentation and usage examples. 

Fixes # (issue)

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

### CODE REVIEW

**Comment by Ayan-Bandyopadhyay** [comment]
@eyurtsev added the dependency just now. 

One question I had was -- how can I run the local version of langchain? 
`poetry install -E all` only seems to install the dependencies. When I import langchain in a project it still uses the pypi version not the local verison.

File: langchain/document_loaders/psychic.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            )
+            return [
+                Document(
+                    page_content=doc["content"],
+                    metadata={"title": doc["title"], "source": doc["uri"]},
+                )
+                for doc in psychic_docs
+            ]
+        except Exception as e:
+            print(e)
```
Comment (hwchase17) [inline langchain/document_loaders/psychic.py:40]: may be better just throw and let the user handle? seems a bit opaque to handle in here

**Review by hwchase17 — COMMENTED** [review]
minor nit but looks good to me

**Review by eyurtsev — APPROVED** [review]

File: docs/ecosystem/psychic.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Psychic
+
+This page covers how to use [Psychic](https://www.psychic.dev/) within LangChain.
```
[the remark this answers is not in the document]
Comment (Ayan-Bandyopadhyay) [reply docs/ecosystem/psychic.md:3]: Good callout! I'll add some docs.

File: langchain/document_loaders/psychic.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+
+class PsychicLoader(BaseLoader):
+    """Loader that loads documents from Psychic.dev."""
+
+    def __init__(self, api_key: str, connector_id: str, connection_id: str):
+        """Initialize with API key, connector id, and connection id."""
+        self.api_key = api_key
```
Comment (dev2049) [inline langchain/document_loaders/psychic.py:13]: do we need to store api key?

File: langchain/document_loaders/psychic.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            )
+            return [
+                Document(
+                    page_content=doc["content"],
+                    metadata={"title": doc["title"], "source": doc["uri"]},
+                )
+                for doc in psychic_docs
+            ]
+        except Exception as e:
+            raise RuntimeError("Error loading documents from Psychic.dev: {}".format(e))
```
Comment (dev2049) [inline langchain/document_loaders/psychic.py:40]: not sure this error wrapping adds much, imo could just remove the try/except

File: langchain/document_loaders/psychic.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    def __init__(self, api_key: str, connector_id: str, connection_id: str):
+        """Initialize with API key, connector id, and connection id."""
+        self.api_key = api_key
+
+        try:
+            from psychicapi import ConnectorId, Psychic  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "`psychicapi` package not found, please run " "`pip install psychicapi`"
```
Comment (dev2049) [inline langchain/document_loaders/psychic.py:19]: combine strings

**Review by dev2049 — COMMENTED** [review]
could we add a simple unit test?

**Comment by Ayan-Bandyopadhyay** [comment]
@dev2049 Just updated the PR w/ unit tests and the changes you mentioned

**Pull request merged**